src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationService.kt

```diff
@@ -149,24 +150,24 @@ class ActivitiesMigrationService(
         null,
       )
     }
-}
 
-private fun GetActivityResponse.toActivityMigrateRequest(requestedStartDate: LocalDate): ActivityMigrateRequest = ActivityMigrateRequest(
-  programServiceCode = programCode,
-  prisonCode = prisonId,
-  startDate = maxOf(startDate, requestedStartDate),
-  endDate = endDate,
-  capacity = max(1, capacity),
-  description = description,
-  payPerSession = ActivityMigrateRequest.PayPerSession.valueOf(payPerSession),
-  runsOnBankHoliday = !excludeBankHolidays,
-  internalLocationCode = internalLocationCode,
-  internalLocationId = internalLocationId,
-  internalLocationDescription = internalLocationDescription,
-  scheduleRules = scheduleRules.toNomisScheduleRules(),
-  payRates = payRates.map { it.toNomisPayRate() },
-  outsideWork = outsideWork,
-)
+  private suspend fun GetActivityResponse.toActivityMigrateRequest(requestedStartDate: LocalDate): ActivityMigrateRequest = ActivityMigrateRequest(
+    programServiceCode = programCode,
+    prisonCode = prisonId,
+    startDate = maxOf(startDate, requestedStartDate),
+    endDate = endDate,
+    capacity = max(1, capacity),
+    description = description,
+    payPerSession = ActivityMigrateRequest.PayPerSession.valueOf(payPerSession),
+    runsOnBankHoliday = !excludeBankHolidays,
+    dpsLocationId = internalLocationId.toDpsLocationId(),
```

Comment:
The main change is to get the DPS location from the mapping service and stop sending the NOMIS location details. This will allow activities to remove internal location from the API and make dps location non-nullable.